### PR TITLE
Add variable.item()

### PR DESCRIPTION
**Pull request by crcrpar**
This PR adds `item()` method to `chainer.Variable`.

### CODE REVIEW

**Comment by niboshi** [comment]
What's the motivation of this method? 🤔 Isn't `var.array.item()` enough?

**Comment by crcrpar** [comment]
main motivation is to provide users with one concrete and simple way to convert variables with 1 value such as variables representing loss to python scalar.

IMO, currently, it's a bit demanding and not clear to do this. ref: DictSummary

**Comment by niboshi** [comment]
OK, thank you. I think we should

* mention that it may incur host-device synchronozation,
* mention that ChainerX is not supported,
* raise `NotImplementedError` with appropriate message for ChainerX.

File: chainer/variable.py [hunk 1]
```diff
@@ -1424,6 +1424,18 @@ def backward(self, retain_grad=False, enable_double_backprop=False,
         with chainer.using_config('enable_backprop', enable_double_backprop):
             _backprop_to_all([self], retain_grad, loss_scale)
 
+    def item(self):
+        """Converts the variable with one element to a Python scalar.
+
+        This will incur host-device synchronization. Note that ChainerX is not
```
Comment (toslunar) [inline chainer/variable.py]: ChainerX is supported :smile:

Comment (crcrpar) [reply chainer/variable.py]: Thanks. fixed

File: tests/chainer_tests/test_variable.py [hunk 2]
```diff
@@ -2329,6 +2329,53 @@ def test_backward_chainerx(self):
         self.check_backward(chainerx.array(self.x))
 
 
+@testing.parameterize(*testing.product({
+    'shape': [(), (1,), (1, 1), (1, 1, 1), (2,), (2, 3)],
```
Comment (toslunar) [inline tests/chainer_tests/test_variable.py]: Why don't you test empty shapes (`x.size == 0`)?

Comment (crcrpar) [reply tests/chainer_tests/test_variable.py]: I missed them. Thank you.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
The Jenkins failures are unrelated to the PR.

**Pull request merged**